services/ad-tech/src/lib/template-utils.ts

```diff
@@ -11,46 +11,67 @@
  limitations under the License.
  */
 
-import {PORT, EXTERNAL_PORT} from '../lib/constants.js';
-import {HOSTNAME, SHOP_HOST} from '../lib/constants.js';
-import {DSP_HOST, DSP_A_HOST, DSP_B_HOST} from '../lib/constants.js';
-import {SSP_HOST, SSP_A_HOST, SSP_B_HOST} from '../lib/constants.js';
-import {DSP_DETAIL, DSP_A_DETAIL, DSP_B_DETAIL} from '../lib/constants.js';
-import {SSP_DETAIL, SSP_A_DETAIL, SSP_B_DETAIL} from '../lib/constants.js';
+import {
+  ADVERTISER_CONTEXTUAL,
+  DSP_A_DETAIL,
+  DSP_A_HOST,
+  DSP_B_DETAIL,
+  DSP_B_HOST,
+  DSP_DETAIL,
+  DSP_HOST,
+  EXTERNAL_PORT,
+  HOME_HOST,
+  HOSTNAME,
+  PORT,
+  SHOP_HOST,
+  SSP_A_DETAIL,
+  SSP_A_HOST,
+  SSP_B_DETAIL,
+  SSP_B_HOST,
+  SSP_DETAIL,
+  SSP_HOST,
+} from '../lib/constants.js';
 
 /** Returns variables for use in the ad template. */
 export const getAdTemplateVariables = (requestQuery: any) => {
   // Initialize template variables.
-  const advertiser = requestQuery.advertiser || HOSTNAME;
+  const advertiser = requestQuery.advertiser?.toString() || HOSTNAME!;
+  const registerSourceUrl = new URL(
+    `https://${HOSTNAME}:${EXTERNAL_PORT}/register-source`,
+  );
+  registerSourceUrl.searchParams.append('advertiser', advertiser);
+  if (ADVERTISER_CONTEXTUAL === advertiser) {
+    return {
+      title: `Contextual ads from ${ADVERTISER_CONTEXTUAL}`,
+      destination: new URL(`https://${HOME_HOST}:${EXTERNAL_PORT}`).toString(),
```

Comment:
why not a dummy advertiser page ?
(HOME is only here for hosting documentation, I would like to avoid mixing the documentation with demos components)

Reply:
Done, switching to TRAVEL_HOST